src/pages/NewRequest.js

```diff
@@ -1,9 +1,189 @@
+
+import {
+  Administrative,
+  AdministrativeFm,
+  Paid,
+  SickNoDoc,
+  SickWithDoc,
+  Social,
+  Study,
+} from '../components/Leaves/index';
+
+//ForceMajeureAdministrativeLeave = 1,
+// AdministrativeUnpaidLeave = 2,
+// SocialLeave = 3,
+// SickLeaveWithoutDocuments = 4,
+// SickLeaveWithDocuments = 5,
+// StudyLeave = 6,
+// PaidLeave = 7
+
+let prManagers = [
+  'Anney Kirillova',
+  'Bill Ivanov',
+  'Reygar Smith',
+  'Kirill Retushev',
+  'Yevheniy Enderov',
+  'Bill Ivanova',
+  'John Reygar',
+  'John Killian',
+  'John Lothbrok',
+  'John Kirev',
+  'John Kirov',
+];
+
+function NewRequest({ isOpen, onClose }) {
+  const [context, setContext] = useContext(Context);
+  const [leaveType, setLeaveType] = useState(7);
+  const [isSendingRequest, setRequestSending] = useState(false);
+  const [comment, setComment] = useState('');
+  const [fromDate, setFromDate] = useState(null);
+  const [toDate, setToDate] = useState(null);
+  const [pmanager, setPManager] = useState(['']);
+  const [duration, setDuration] = useState(2);
+
+  const handleComment = (e) => {
+    setComment(e.target.value);
+  };
+
+  const handleFromDate = (date) => {
+    setFromDate(date);
+  };
+
+  const handleToDate = (date) => {
+    setToDate(date);
+  };
+
+  const handleManagers = (array) => {
+    setPManager(array);
+  };
+
+  const handleSendRequest = async () => {
+    setRequestSending(true);
+
+    //real request
+    // await postNewRequest({
+    //   TypeId: leaveType,
+    //   StartDate: fromDate,
+    //   EndDate: toDate,
+    //   ReviewersIds: pmanager,
+    //   HasAccountingReviewPassed: false, //лишнее
+    //   Comment: comment,
+    //   State: 'New',
+    //   DurationId: duration,
+    //   UserId: context.userId,
+    // })
+    //   .then(({ data }) => console.log(data))
+    //   .catch((err) => {
+    //     console.log(err);
+    //     console.log(err.response.data);
+    //   });
+
+    await axios
+      .post('https://url')
+      .then((data) => {})
+      .catch((err) => {});
+    setRequestSending(false);
+  };
+
+  const handleDuration = (duration) => {
+    setDuration(duration);
+  };
+
+  useEffect(() => {
+    async function getAllData() {
+      await getAllManagers().then(({ data }) => {
+        //закомментировал для корректной работы с локальным беком
+        //prManagers = data;
+      });
+    }
+    getAllData();
+  }, []);
+
+  const typeProps = {
+    prManagers: prManagers,
+    isSendingRequest: isSendingRequest,
+    comment: comment,
+    changeComment: handleComment,
+    fromDate: fromDate,
+    changeFromDate: handleFromDate,
+    toDate: toDate,
+    changeToDate: handleToDate,
+    managers: pmanager,
+    changeManagers: handleManagers,
+    duration: duration,
+    changeDuration: handleDuration,
+  };
+
+  const renderLeaveBody = [
+    null,
+    { title: 'Administrative force majeure leave', comp: <AdministrativeFm {...typeProps} /> },
```

Comment:
add id from back-end, use it as key for vacation type dropdown

Reply:
Done.